### ISSUE TITLE
Mixed authentication with client certificate breaks auth on shared custom domain API

### ISSUE THREAD

**Issue by janeves** [TykTechnologies/tyk#2625]
**Branch/Environment/Version**
- Branch/Version: [reproduced in 2.9]
- Environment: [on-prem]

**Describe the bug**
When attempting to enable mixed authentication (in this case bearer token and client certificate) with two APIs that share the same custom domain and SSL certificate, enabling the token and client certificate auth on the first API breaks the auth on the second API without making any changes to it.

**Reproduction steps**

1.  Create two APIs
2. Create a custom domain and SSL for custom domain
3. Attach domain and SSL to both APIs
4. Enable auth token on both APIs and create keys for both (works fine at this point)
5. Enable mixed authentication (token and client certificate) on API1
6. Attempting to access API2 after the change of API1 will return an error

**Expected behavior**
API2 should continue to work normally or also allow mixed auth (token and client cert) independently of API1

**Additional context**
Customer is attempting to enable mixed authentication (token and client cert) on APIs that share the same custom domain with SSL

**Comment by buger** [comment]
What exact issue? 

Are you talking about accessing via API or Web browser?

When you enable mutual tls or certificate check on domain, it will mean that server should ask client for client certificate. So if APIs share the same domain, and only one require cert check, user will be asked for client certificate in all cases. 

When you access via curl or similar, it usually work as expected, but when you access such API via browser it can show you the window asking to select certificate.

In other words needs more details.

**Comment by christtyk** [comment]
@janeves FYI

**Comment by christtyk** [comment]
@buger this is the bell ticket for 4214. Adding to 2.9.3

**Comment by buger** [comment]
I was able to confirm and fix the bug. 
The general idea is that if we had 2 apis with non-empty custom domain field set, and only ONE of them has mutual tls enabled, it was enabling it for all APIs on this domain.

After the fix, it continues requesting a client certificate, but does not validate it on TLS level, istead falling back to HTTP certificate check.

**Comment by maciejwojciechowski** [comment]
need to reopen. this works but only if I modify the api created first. Otherwise it does not work.
So this scenario works:
1. add api1
2. add api2
3. modify api1 to be mixed and use mtls
> api2 works fine without cert

but if you change like this
1 & 2 the same
3. modify **api2** to be mixed and use mtls
> then both apis require cert as previously.

**Comment by buger** [comment]
Can you attach APIs which replicate the issue?

**Comment by maciejwojciechowski** [comment]
[api.txt](https://github.com/TykTechnologies/tyk/files/4041575/api.txt)
[api2.txt](https://github.com/TykTechnologies/tyk/files/4041576/api2.txt)

in gateway logs I see
```
2020/01/09 18:03:03 http: TLS handshake error from 127.0.0.1:63583: tls: client didn't provide a certificate
```

And in postman I have `Could not get any response`

which is also different from scenario 1 (where I do get response that certificate is missing)

**Comment by maciejwojciechowski** [comment]
still have the same odd behavior. if you modify api2 to use mTLS it is broken (opposite to when you modify api1 - in this case it works as expected)

**Comment by maciejwojciechowski** [comment]
verified
